forwarder.go

```diff
@@ -277,35 +151,19 @@ func writeJSON(eventlist []string) string {
 	return jsonDoc
 }
 
-func writeToLogChan(eventlist []string, logChan chan string) {
+func processAndEnqueue(eventlist []string) {
 	if len(eventlist) > 0 { //only attempt delivery if eventlist contains elements
 		jsonSTRING := writeJSON(eventlist)
-		t := metrics.GetOrRegisterTimer("post.queue.latency", metrics.DefaultRegistry)
-		t.Time(func() {
-			//log.Printf("Sending document to channel: %v", jsonSTRING)
-			logChan <- jsonSTRING
-		})
+		logDispatch.Enqueue(jsonSTRING)
```

Comment:
I would try refactoring forwarder.go 

I would make a StringProcessorService or some other name you consider, that would contain a dispatcher, this processAndEnqueue method and the other 2 methods that get eventlist as param, and extract that big for loop from the main function and put it there in a process() function or something like that.

I would try clearing the global vars from this file and give them as params in this new processor or move them in the new file (whatever is specific for it).